### PR TITLE
improve column access speed for sparse matrices

### LINKED ISSUE #719: slow querying of gene expression values
1M+ cell dataset. `.X` is a sparse array of uint16 values.

I want to visualize the expression of a specific gene. Autosuggest is snappy and helps me type "Vip". I hit "enter" and get a spinning wheel for 4m05s before the histogram appears. Once it appears, the UI snappy again.

### PR DESCRIPTION
**Pull request by bkmartinjr**
This is a partial fix for performance issues with sparse matrices.   Change adds sparse-aware column slicing to the access route which accesses the AnnData.X matrix, and an FAQ note that CSC or dense matrices are preferred for performance.

Fixes #719

### CODE REVIEW

File: server/app/scanpy_engine/scanpy_engine.py [hunk 1]
```diff
@@ -313,10 +330,9 @@ def data_frame_to_fbs_matrix(self, filter, axis):
             raise FilterError("filtering on obs unsupported")
 
         # Currently only handles VAR dimension
-        X = self.data._X
-        if var_selector is not None:
-            X = X[:, var_selector]
-        return encode_matrix_fbs(X, col_idx=np.nonzero(var_selector)[0], row_idx=None)
+        X = self.slice_columns(self.data._X, var_selector)
+        m = encode_matrix_fbs(X, col_idx=np.nonzero(var_selector)[0], row_idx=None)
```
[the remark this answers is not in the document]
Comment (bkmartinjr) [reply server/app/scanpy_engine/scanpy_engine.py]: Oops - this was debugging detritus.  Will fix.

File: docs/faq.md [hunk 2]
```diff
@@ -40,6 +40,14 @@ Currently this is not supported directly, but you should be able to do this your
 - `.X` is used to display expression (histograms, scatterplot & colorscale) and to compute differential expression
 - `.obsm` is used for layout
 
+#### I have a BIG dataset - how can I make cellxgene run as fast as possible?
```
[the remark this answers is not in the document]
Comment (bkmartinjr) [reply docs/faq.md]: What units make sense?  The underlying issue is the size in memory - closely related to the h5ad file size.   There are a number of factors that influence this:
* total number of genes & cells, and the matrix density
* format used to encode the file (dense vs sparse, dtype, etc)

The most accurate statement would be something like - if you have a dataset which requires GBs to store on disk, then it is big.    Is that useful in this context?

[the remark this answers is not in the document]
Comment (csweaver) [reply docs/faq.md]: Yes, I think file size is a valid source of measurement for this community.

[the remark this answers is not in the document]
Comment (bkmartinjr) [reply docs/faq.md]: OK, this has been reworded/expanded/etc.

[the remark this answers is not in the document]
Comment (csweaver) [reply docs/faq.md]: looks good!

**Pull request merged**